frontend/plugincompatibility.lua

```diff
@@ -0,0 +1,367 @@
+--[[--
+Handles plugin version compatibility checks and load overrides.
+
+This module checks if a plugin is compatible with the current KOReader version
+based on the compatibility field in the plugin's _meta.lua file. It also manages
+user overrides for loading incompatible plugins.
+
+Settings are stored in a dedicated file via PluginCompatibilitySettings.
+Access settings directly through the `settings` field.
+
+To develop this feature, set KODEV_ENABLE_INCOMPATIBLE_PLUGIN env variable
+to any value to make hello.koplugin appear as an incompatible plugin.
+Check ./plugins/hello.koplugin/_meta.lua to see how this works.
+
+@usage
+    local PluginCompatibility = require("plugincompatibility")
+    local compatibility = PluginCompatibility:new()
+
+    -- Check compatibility (static method)
+    local is_compatible, reason, message = PluginCompatibility.checkCompatibility(plugin_meta)
+
+    -- Manage prompts and overrides via settings
+    compatibility.settings:markAsPrompted("myplugin", "1.0")
+    compatibility.settings:setLoadOverride("myplugin", "1.0", "always")
+
+    -- Flush settings when done
+    compatibility.settings:flush()
+
+    -- Clean up old settings (keep current and 2 previous versions)
+    compatibility.settings:purgeOldVersionSettings(3)
+]]
+
+local InfoMessage = require("ui/widget/infomessage")
+local Menu = require("ui/widget/menu")
+local PluginCompatibilitySettings = require("plugincompatibilitysettings")
+local T = require("ffi/util").template
+local UIManager = require("ui/uimanager")
+local Version = require("frontend/version")
+local _ = require("gettext")
+local logger = require("logger")
```

Comment:
Please abide by https://github.com/koreader/koreader-base/wiki/Coding-style#order-of-requires